src/rp2_common/hardware_i2c/i2c.c

```diff
@@ -67,8 +67,8 @@ uint i2c_set_baudrate(i2c_inst_t *i2c, uint baudrate) {
 
     // TODO there are some subtleties to I2C timing which we are completely ignoring here
     uint period = (freq_in + baudrate / 2) / baudrate;
```

Comment:
Thanks for this. Do you also understand why `period = (freq_in + baudrate / 2) / baudrate;` rather than simply `period = freq_in / baudrate;` ? I was puzzled by this.

Reply:
maybe it's just rounding.

Reply:
Yes, it's rounding for int/uint: (dividend + divisor / 2) / divisor